osm2gtfs/core/osm_connector.py

```diff
@@ -170,14 +170,7 @@ def get_routes(self, refresh=False):
             if line is None:
                 continue
 
-            # Make sure route_id (ref) number is not already taken
-            if line.route_id in self.routes:
```

Comment:
I think this check for unique `ref` is still valid with the introduced changes. Probably should be moved to the creators now.

Reply:
Hum, I disagree : I actually have two different bus lines ([Stigo 18](https://www.openstreetmap.org/relation/6957343) / [Seine-et-Marne Express 18](https://www.openstreetmap.org/relation/6957245)) with the same ref that run in my neighborhood. In Abidjan, we even have perfect duplicates, where one agency operates multiples lines with the same number (for instance the 13 one : https://www.openstreetmap.org/relation/10270789 vs https://www.openstreetmap.org/relation/10266011)
So if we want to keep this rules, it definitly should be in the creators and not in the core.

And I think we should handle duplicates in OSM instead of doing some magic to make them disappear in osm2gtfs.

But this is a different issue, so I can remove it from this PR if you prefer.

Reply:
It looks like we have the common the idea, that the standard way of populating the GTFS `route_id` from OSM's `ref` tag - as this PR introduces. Highlighting here the [GTFS best practices](https://gtfs.org/best-practices/#routestxt) regarding the `route_id`:

> All trips on a given named route should reference the same route_id.
> * Different directions of a route should not be separated into different route_id values.
> * Different spans of operation of a route should not be separated into different route_id values. i.e. do not create different records in routes.txt for “Downtown AM” and “Downtown PM” services).

So, we have the situation that in OSM there can be duplicated `ref` values, as you line out in your comment, @nlehuby. And in GTFS they should be grouped to one entry. While I do think we have to build this bridge in osm2gtfs, I agree that the current logic of just ignoring the repeated occurrence is not addressing all use cases. But it is better than just dropping it entirely. 

I suggest we keep this logic in this PR, but move it to the creators. Then we can consider improving the logic in a separate issue, conversation and PR.

Reply:
> I suggest we keep this logic in this PR, but move it to the creators
it seems to me that no creators is ultimately using this logic. None of them have `ref` duplicates

Reply:
Yes, because the data is good enough. The checks make sure that even if people add a duplicated 'ref' in OSM that osm2gtfs produces still a valid GTFS. As said, I think this can be improved, but taking it out, seems to be a step backwards and towards more inconsistent GTFS files.

Reply:
> And I think we should handle duplicates in OSM instead of doing some magic to make them disappear in osm2gtfs.

Yes, I deeply agree with that, and this is why I'm against dropping them.

In the case there is a duplicated `ref` in OSM data, but in GTFS there shall be only one unique `route_id`.

What happens with the checks:

* It generates a valid GTFS (omitting some routes, so incomplete).
* It displays an error message, so that people can:
  - Either fix it in OSM with one `route_master` for the different routes.
  - Use a custom creator to do their own logic.

What happens without the checks:

* It generates an invalid GTFS (with duplicated `route_id`)
* No warning message is presented to the user.

Reply:
ok, I just understood your point.
I had forgotten that OSM `ref` tag was also used to create GTFS `route_id` (I usually use osm relation id as route_id) and I thought the debate was mostly about the `route_short_name` field. 
You are right, without the check, we have an invalid GTFS.